app/store/middleware/onChangedSelectedHealthAuthorities.ts

```diff
@@ -0,0 +1,23 @@
+import { Dispatch, Middleware, AnyAction, MiddlewareAPI } from 'redux';
+import IntersectService from '../../services/IntersectService';
+
+const onChangedSelectedHealthAuthorities: Middleware<Dispatch> = (
+  store: MiddlewareAPI,
+) => (next: Dispatch<AnyAction>) => (action: AnyAction): unknown => {
+  const {
+    healthcareAuthorities: { selectedAuthorities: stateBefore },
+  } = store.getState();
+
+  const result = next(action);
+
+  const {
```

Comment:
Can these be wrapped in a try to ensure the chain will  always complete (return result)